libraries/browser-tracker-core/src/tracker/index.ts

```diff
@@ -713,6 +716,23 @@ export function Tracker(
       return state.pageViewId;
     }
 
+    /**
+     * Safe function to get `tabId`.
+     * Generates it if it is not yet initialized. Shared between trackers.
+     */
+    function getTabId() {
```

Comment:
I'm wondering what should happen in case the `stateStorageStrategy` is `none` (and anonymous tracking is off) – would it make sense to use the in-memory `SharedState` storage as we do for the `pageViewId`? It complicates the logic to store the value in multiple places so I'm not sure if it's worth it. But I think it'd make sense because otherwise, the `pageViewId` (also the `domainUserId` since it is also cached in memory) would be consistent for some events tracked with the same in-memory tracker instance, but the `tabId` would be changing on every event – that could be quite confusing for users.

Reply:
_Went over this as well as we discussed offline._
I totally see the reasoning you stated above. 
What I am thinking is that storing the `tabId` in memory will make it more **consistent** but will not make it more **reliable** if storage is not allowed. This attribute, in my mind at least, is a tightly coupled notion to the browser storage and persistence of the same tab. 
That is the tradeoff I am thinking about for adding it in the `SharedState` and the logic that would be required.

Reply:
That's a good point! I can see that storing it in memory gives an illusion of reliability in the data. On the other hand, we made the trade-off in other places to store other identifiers in memory and I'm also nervous about having different lifespans for different identifiers. Also there might be some value in the in-memory `tabId` for some client-side JS apps (or maybe not, I don't really know what use cases it addresses).

Remind me please again why we decided to generate a new UUID for the `tabId` even if we can't persist it instead of returning `null`? I know we talked about this on the call, but already forgot the reasoning. Returning `null` might be a way to avoid setting the user expectations for the same lifespan compared to `pageViewId` for instance when storage is `none`.

Reply:
From the offline discussion:
- We wil reform the schema to accept null values for the tabId.
- In the cases of anonymous tracking, no storage and access denial on session storage we will return `null`.
- When the web_page context is deactivated and the `getTabId()` API is called, we again will return null.

Reply:
I appreciate the link of this feature to browser storage, however conceptually, it seems strange to have a `null` `tabId` - since all events from a browser must be done from within a tab.

I think when we consider anonymous tracking, having a `null` id makes sense. Perhaps we should introduce a flag like we have done for session tracking - so users can keep the tabId even when anonymous tracking is on but by default we set it to `null`. I think this is reason enough to allow `null` in the schema.

What do we do for session id when the `stateStorageStrategy` is `none`? I feel like the tabId should probably behave similarly to that? Unless however we've implemented is that it rotates all the time (which is pointless but I can't remember), in which case `null` makes more sense. As I type this I'm imagining `null` does actually make sense, especially for non-SPA, as if there's no storage then we can't keep track of the tab between page loads.